ambry-store/src/main/java/com.github.ambry.store/StatsManager.java

```diff
@@ -117,30 +122,28 @@ void publish(StatsWrapper statsWrapper) throws IOException {
   }
 
   /**
-   * Collect and aggregate quota stats from all given {@link Store}s by fetching stats from them sequentially via the
-   * {@link StorageManager}.
-   * @param partitionIds a {@link Set} of {@link PartitionId}s representing a set of {@link Store}s to be fetched
-   * @return a {@link Pair} where the first element is the result in the form of a {@link StatsSnapshot} and the
-   * second element is the number of {@link Store}s that were skipped (either unreachable or an error has occurred).
+   * Fetch and aggregate stats from a given {@link Store}
+   * @param aggregatedSnapshot the {@link StatsSnapshot} to hold the aggregated result
+   * @param partitionId specifies the {@link Store} to be fetched from
+   * @return
    */
-  Pair<StatsSnapshot, List<String>> collectAndAggregate(Set<PartitionId> partitionIds) {
-    StatsSnapshot aggregatedStatsSnapshot = new StatsSnapshot(0L, null);
-    List<String> unreachableStores = new ArrayList<>();
-    for (PartitionId partitionId : partitionIds) {
-      Store store = storageManager.getStore(partitionId);
-      if (store != null) {
-        try {
-          StatsSnapshot statsSnapshot = store.getStoreStats().getStatsSnapshot();
-          aggregate(aggregatedStatsSnapshot, statsSnapshot);
-        } catch (StoreException e) {
-          logger.error("Store exception thrown when getting stats for partitionId: " + partitionId.toString(), e);
-          unreachableStores.add(partitionId.toString());
-        }
-      } else {
-        unreachableStores.add(partitionId.toString());
+  boolean collectAndAggregate(StatsSnapshot aggregatedSnapshot, PartitionId partitionId) {
+    long fetchAndAggregatePerStoreStartTimeMs = time.milliseconds();
+    Store store = storageManager.getStore(partitionId);
+    if (store == null) {
+      return false;
+    } else {
+      StatsSnapshot statsSnapshot = null;
+      try {
+        statsSnapshot = store.getStoreStats().getStatsSnapshot();
+      } catch (StoreException e) {
+        logger.error("Store exception thrown when getting stats for partitionId: {}", partitionId, e);
+        return false;
```

Comment:
Ok looks like the caller doesn't but that is probably where you should handle all exceptions (instead of doing it in multiple places)